03-Custom-Actions-In-Python/06-Improving-Performance.ipynb

```diff
@@ -0,0 +1,421 @@
+{
```

Comment:
<p>I find the General Guidelines paragraph awkward, especially the introductory paragraph. I think it could be reworded to be clearer. The whole section may also benefit from being shortened - possibly even to a bulleted list. Start off by stating that users should profile their code to find the bottlenecks, and maybe even link to a good tool to do so. Then transition to the remaining paragraphs by stating that these are common bottlenecks in custom actions and here are ways to improve performance of each bottlenecks. </p><p> </p><p>Something like:</p><p>* <code>State.snapshot</code>  is expensive: Use <code>cpu_local_snapshot</code>  or <code>gpu_local_snapshot</code></p><p> </p><p>Avoid judgemental statements like "easiest and most obvious". </p><p> </p><p>I still don't think that Cython/numba/C extensions are appropriate to mention here. Users that are aware of these methods will be knowledgeable enough to apply them.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/hoomd-examples/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>